biogeochem/EDPatchDynamicsMod.F90

```diff
@@ -489,24 +489,37 @@ subroutine spawn_patches( currentSite, bc_in)
                                              ! for both woody and grass species
     real(r8) :: leaf_m                       ! leaf mass during partial burn calculations
     logical  :: found_youngest_primary       ! logical for finding the first primary forest patch
+    integer  :: min_nocomp_pft, max_nocomp_pft, i_nocomp_pft
     !---------------------------------------------------------------------
 
     storesmallcohort => null() ! storage of the smallest cohort for insertion routine
     storebigcohort   => null() ! storage of the largest cohort for insertion routine 
 
-    ! calculate area of disturbed land, in this timestep, by summing contributions from each existing patch. 
-    currentPatch => currentSite%youngest_patch
-
-    site_areadis_primary = 0.0_r8
-    site_areadis_secondary = 0.0_r8    
+    if (hlm_use_nocomp .eq. itrue) then
+       min_nocomp_pft = 0
+       max_nocomp_pft = maxpft
+    else
+       min_nocomp_pft = fates_unset_int
+       max_nocomp_pft = fates_unset_int
+    endif
 
     ! zero the diagnostic disturbance rate fields
     currentSite%disturbance_rates_primary_to_primary(1:N_DIST_TYPES) = 0._r8
     currentSite%disturbance_rates_primary_to_secondary(1:N_DIST_TYPES) = 0._r8
     currentSite%disturbance_rates_secondary_to_secondary(1:N_DIST_TYPES) = 0._r8
 
+    nocomp_pft_loop: do i_nocomp_pft = min_nocomp_pft,max_nocomp_pft
+
+    ! calculate area of disturbed land, in this timestep, by summing contributions from each existing patch. 
+    currentPatch => currentSite%youngest_patch
+
+    site_areadis_primary = 0.0_r8
+    site_areadis_secondary = 0.0_r8    
+
     do while(associated(currentPatch))
 
+       cp_nocomp_matches_1: if ( hlm_use_nocomp .eq. ifalse .or. &
```

Comment:
where the do loops have labels they include the word 'loop' in them, so I feel like the if statement labels might be easier for people to parse if they have 'if' in their name.

Reply:
Yeah I wasn't very consistent in how I did this.  It would probably be easier to read if each label does have an 'if' or 'loop' or something like that in it.  So will edit to try to be more consistent.